tests/functional_tests/ffront_tests/test_type_deduction.py

```diff
@@ -0,0 +1,246 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import pytest
+
+from functional.common import Dimension
+from functional.ffront import field_operator_ast as foast
+from functional.ffront.fbuiltins import Field, float64, int64
+from functional.ffront.foast_passes.type_deduction import FieldOperatorTypeDeductionError, TypeInfo
+from functional.ffront.func_to_foast import FieldOperatorParser
+
+
+def type_info_cases():
+    return [
+        (
+            None,
+            {
+                "is_complete": False,
+                "is_any_type": True,
+                "constraint": None,
+                "is_field_type": False,
+                "is_scalar": False,
+                "is_arithmetic_compatible": False,
+                "is_logics_compatible": False,
+            },
+        ),
+        (
+            foast.DeferredSymbolType(constraint=None),
+            {
+                "is_complete": False,
+                "is_any_type": True,
+                "constraint": None,
+                "is_field_type": False,
+                "is_scalar": False,
+                "is_arithmetic_compatible": False,
+                "is_logics_compatible": False,
+            },
+        ),
+        (
+            foast.DeferredSymbolType(constraint=foast.ScalarType),
+            {
+                "is_complete": False,
+                "is_any_type": False,
+                "constraint": foast.ScalarType,
+                "is_field_type": False,
+                "is_scalar": True,
+                "is_arithmetic_compatible": False,
+                "is_logics_compatible": False,
+            },
+        ),
+        (
+            foast.DeferredSymbolType(constraint=foast.FieldType),
+            {
+                "is_complete": False,
+                "is_any_type": False,
+                "constraint": foast.FieldType,
+                "is_field_type": True,
+                "is_scalar": False,
+                "is_arithmetic_compatible": False,
+                "is_logics_compatible": False,
+            },
+        ),
+        (
+            foast.ScalarType(kind=foast.ScalarKind.INT64),
+            {
+                "is_complete": True,
+                "is_any_type": False,
+                "constraint": foast.ScalarType,
+                "is_field_type": False,
+                "is_scalar": True,
+                "is_arithmetic_compatible": True,
+                "is_logics_compatible": False,
+            },
+        ),
+        (
+            foast.FieldType(dims=Ellipsis, dtype=foast.ScalarType(kind=foast.ScalarKind.BOOL)),
+            {
+                "is_complete": True,
+                "is_any_type": False,
+                "constraint": foast.FieldType,
+                "is_field_type": True,
+                "is_scalar": False,
+                "is_arithmetic_compatible": False,
+                "is_logics_compatible": True,
+            },
+        ),
+    ]
+
+
+@pytest.mark.parametrize("symbol_type,expected", type_info_cases())
```

Comment:
:+1:  Please remind other people to follow the same pattern when the number of cases in `parametrize` grows too much. I should've refactor the `symbol_makers` tests in this way (but the reviewer didn't remind me about it :rofl: )

Reply:
I only came up with this pattern afterwards. At the time I didn't see a way around it but it started me thinking...

Reply:
:+1:  We used it already in the past (with eve, think) but then kind of forgot about it.